java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -914,14 +931,31 @@ public void visitMethodReference(MethodReferenceTree methodReferenceTree) {
       if("new".equals(methodReferenceTree.method().name())) {
         resolution = resolve.findMethod(methodEnv, getType(methodReferenceTree.expression()), "<init>", samMethodArgs);
         associateReference(methodReferenceTree.method(), resolution.symbol());
+        return;
       } else {
         resolveMethodSymbol(methodReferenceTree.method(), methodEnv, samMethodArgs, ImmutableList.<JavaType>of());
       }
+      JavaType methodRefType = (JavaType) methodRefTree.symbolType();
+      if (methodRefTree.symbolType().isUnknown() || methodRefType.isTagged(JavaType.DEFERRED)) {
```

Comment:
why not use `methodRefType` instead of calling `methodRefTree.symbolType()` again?